devito/function.py

```diff
@@ -1189,7 +1189,7 @@ def __shape_setup__(cls, **kwargs):
             shape = (glb_npoint[grid.distributor.myrank],)
         return shape
 
-    @property
+    @cached_property
```

Comment:
as discussed already, we cannot afford these yet (not unless we ``del`` them each time the coordinates are written to)